framework/include/utils/InputParameters.h

```diff
@@ -278,19 +278,42 @@ class InputParameters : public Parameters
   void addCoupledVar(const std::string &name, const Real value, const std::string &doc_string);
 
   /**
+   * This method adds a coupled variable name pair.  The parser will look for variable
+   * name pair in the input file and can return a reference to the storage location
+   * for the coupled variable if found.
+   *
+   * This version of the method will build a vector if the given the base_name and num_name parameters exist
+   * in the input file:
+   *   e.g.
+   *   [./foo]
+   *     ...
+   *     some_base = base_
+   *     some_num  = 5
+   *   [../]
+   *
+   *   # The coupling parameter will be passed this vector: "base_0 base_1 base_2 base_3 base_4"
+   */
+  void addCoupledVar(const std::string &name, const std::string &base_name, const std::string &num_name, const std::string &doc_string);
```

Comment:
Do we still want to try to find a new name? If so, how about `addCoupledVarByBase` or `addCoupledVarWithBase` or `addCoupledVarGivenABaseNameAndAnIntergerRepresentingHowManyVariablesToCreate`